keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,1129 @@
+    for each are recorded in DeviceResourceRequestAllocationResult, and just 
+    one claim object with allocation `Results` that lists all allocated devices
+    is created for the pod.
+
+The special resource claim object lifecycle is managed by the scheduler and
+garbage collector.
+
+  * It is *created* in a namespace when there is a pod with extended resource
+  request, and the extended resource is advertised by `ResourceSlice` and
+  scheduler has fit the pod to a node with the `ResourceSlice`.
+  * It is *created* by the scheduler dynamic resource plugin during
+    preBind phase. The in-memory one in the assumed cache is created earlier
+    during Reserve phase.
+  * It is *deleted*
+    * either together with the owning pod's deletion.
+    * or by the scheduler dynamic resource plugin during unReserve phase.
+    * or by the scheduler dynamic resource plugin during postFilter phase.
+  * It is *discovered* by the kubelet via `pod.Status.ExtendedResourceClaimStatus`
+  * It is *read* by the kubelet DRA device driver to prepare the devices listed
+    therein when preparing to run the pod.
+
+```go
+type DeviceRequest struct {
+	// Name can be used to reference this request in a pod.spec.containers[].resources.claims
+	// entry and in a constraint of the claim.
+	//
+	// Must be a DNS label.
+	//
+	// +required
+	Name string
+}
+```
+
+To enable the kubelet to map devices back to the containers which requested them,
+the kube-scheduler creates one `DeviceRequest` per extended resource backed by DRA
+per container in the pod. containers can be initContainers, regular containers,
+but cannot be ephemeral containers. The name of the `DeviceRequest` has the form
+"container-%d-request-%d", where the first %d is the index of the container in the pod.
+The second %d is the index of the extended resource inside the container
+resource requests. For example, if the first container in the pod has an
+extended resource backed by DRA which is the 3rd such request in the container,
+then the name of the `DeviceRequest` is "container-0-request-2".
+
+Documenting this naming is merely informational, it is not part of the API.
+The kubelet must not rely on it. Instead, the
+`ContainerExtendedResourceRequest` field below specifies the mapping.
+
+### Pod API
+
+A new field `extendedResourceClaimStatus` is added to Pod's status to track
+the special `ResourceClaim` object created for the extended resource requests
+in the pod. This is needed for kubelet to pass the devices allocated by driver
+to the containers in the pod. containers can be initContainers, regular containers,
+but cannot be ephemeral containers. 
+
+```go
+// PodExtendedResourceClaimStatus is stored in the PodStatus for each extended
+// resource requests backed by DRA. It stores the generated name for 
+// the corresponding special ResourceClaim created by scheduler.
+type PodExtendedResourceClaimStatus struct {
```

Comment:
Can we re-use this field: https://github.com/kubernetes/enhancements/blob/b869b5ce260577c14173a0649ff69e18c5ad3ffd/keps/sig-node/4680-add-resource-health-to-pod-status/README.md?plain=1#L226